[CI] CartesianCentroidIT testSingleValueFieldGetProperty failing

**Build scan:**
https://gradle-enterprise.elastic.co/s/rhbo3mocrcjcg/tests/:x-pack:plugin:spatial:internalClusterTest/org.elasticsearch.xpack.spatial.search.aggregations.metrics.CartesianCentroidIT/testSingleValueFieldGetProperty

**Reproduction line:**
```
./gradlew ':x-pack:plugin:spatial:internalClusterTest' --tests "org.elasticsearch.xpack.spatial.search.aggregations.metrics.CartesianCentroidIT.testSingleValueFieldGetProperty" -Dtests.seed=B790081CCCD10ED9 -Dtests.locale=ms -Dtests.timezone=Asia/Baku -Druntime.java=17
```

**Applicable branches:**
main

**Reproduces locally?:**
Yes

**Failure history:**
https://gradle-enterprise.elastic.co/scans/tests?tests.container=org.elasticsearch.xpack.spatial.search.aggregations.metrics.CartesianCentroidIT&tests.test=testSingleValueFieldGetProperty

**Failure excerpt:**
```
java.lang.AssertionError: Mismatching value for 'y' field of centroid
Expected: a numeric value within <1.4012984643248172E-50> of <1.401298464324817E-45>
     but: <0.0> differed by <1.4012844513401739E-45> more than delta <1.4012984643248172E-50>

  at __randomizedtesting.SeedInfo.seed([B790081CCCD10ED9:4AB0A665F23B21AE]:0)
  at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
  at org.junit.Assert.assertThat(Assert.java:956)
  at org.elasticsearch.search.aggregations.metrics.CentroidAggregationTestBase.assertSameCentroid(CentroidAggregationTestBase.java:165)
  at org.elasticsearch.search.aggregations.metrics.CentroidAggregationTestBase.testSingleValueFieldGetProperty(CentroidAggregationTestBase.java:109)
  at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(NativeMethodAccessorImpl.java:-2)
  at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
  at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
  at java.lang.reflect.Method.invoke(Method.java:568)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:44)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:843)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:490)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:850)
  at java.lang.Thread.run(Thread.java:833)

```

--- a/test/framework/src/main/java/org/elasticsearch/search/aggregations/metrics/CentroidAggregationTestBase.java
+++ b/test/framework/src/main/java/org/elasticsearch/search/aggregations/metrics/CentroidAggregationTestBase.java
@@ -103,26 +103,15 @@ public abstract class CentroidAggregationTestBase extends AbstractGeoTestCase {
         assertThat(global.getAggregations().asMap().size(), equalTo(1));
 
         CentroidAggregation geoCentroid = global.getAggregations().get(aggName());
+        InternalAggregation agg = (InternalAggregation) global;
         assertThat(geoCentroid, notNullValue());
         assertThat(geoCentroid.getName(), equalTo(aggName()));
-        assertThat((CentroidAggregation) ((InternalAggregation) global).getProperty(aggName()), sameInstance(geoCentroid));
+        assertThat((CentroidAggregation) agg.getProperty(aggName()), sameInstance(geoCentroid));
         assertSameCentroid(geoCentroid.centroid(), singleCentroid);
-        assertThat(
-            ((SpatialPoint) ((InternalAggregation) global).getProperty(aggName() + ".value")).getY(),
-            closeTo(singleCentroid.getY(), GEOHASH_TOLERANCE)
-        );
-        assertThat(
-            ((SpatialPoint) ((InternalAggregation) global).getProperty(aggName() + ".value")).getX(),
-            closeTo(singleCentroid.getX(), GEOHASH_TOLERANCE)
-        );
-        assertThat(
-            (double) ((InternalAggregation) global).getProperty(aggName() + "." + coordinateName("y")),
-            closeTo(singleCentroid.getY(), GEOHASH_TOLERANCE)
-        );
-        assertThat(
-            (double) ((InternalAggregation) global).getProperty(aggName() + "." + coordinateName("x")),
-            closeTo(singleCentroid.getX(), GEOHASH_TOLERANCE)
-        );
+        assertSimilarValue(((SpatialPoint) agg.getProperty(aggName() + ".value")).getY(), singleCentroid.getY());
+        assertSimilarValue(((SpatialPoint) agg.getProperty(aggName() + ".value")).getX(), singleCentroid.getX());
+        assertSimilarValue((double) agg.getProperty(aggName() + "." + coordinateName("y")), singleCentroid.getY());
+        assertSimilarValue((double) agg.getProperty(aggName() + "." + coordinateName("x")), singleCentroid.getX());
         assertEquals(numDocs, (long) ((InternalAggregation) global).getProperty(aggName() + ".count"));
     }
 
@@ -153,6 +142,12 @@ public abstract class CentroidAggregationTestBase extends AbstractGeoTestCase {
     /** Override this if the spatial data needs custom normalization (eg. cartesian) */
     protected double normalize(double value) {
         return value;
+    }
+
+    protected void assertSimilarValue(double a, double b) {
+        a = normalize(a);
+        b = normalize(b);
+        assertThat(a, closeTo(b, tolerance(a, b)));
     }
 
     protected void assertSameCentroid(SpatialPoint centroid, SpatialPoint expectedCentroid) {

--- a/x-pack/plugin/spatial/src/internalClusterTest/java/org/elasticsearch/xpack/spatial/search/aggregations/metrics/CartesianCentroidIT.java
+++ b/x-pack/plugin/spatial/src/internalClusterTest/java/org/elasticsearch/xpack/spatial/search/aggregations/metrics/CartesianCentroidIT.java
@@ -71,7 +71,7 @@ public class CartesianCentroidIT extends CentroidAggregationTestBase {
 
     @Override
     protected double tolerance(double a, double b) {
-        return Math.max(Math.abs(a), Math.abs(b)) / 1e5;
+        return Math.max(GEOHASH_TOLERANCE, Math.max(Math.abs(a), Math.abs(b)) / 1e5);
     }
 
     @Override

